### PR TITLE
Refactor CLI error reporting to improve readability

### PR DESCRIPTION
**Pull request by happz**
Added a few empty lines to better separate exceptions in the chain, and while I was there, I refactored the code slightly to separate general exceptions from the special handling of `RunError` ones.

### CODE REVIEW

**Comment by happz** [comment]
I ran into issues with exception reporting while working on the #1790 patch, when things turned south, the output became a bit messy. Therefore adding `multihost` label right away.

File: bin/tmt [hunk 1]
```diff
@@ -1,34 +1,28 @@
...
-def show_exception(exception: BaseException) -> None:
-    """ Show exception in red, handle exception chain """
-    click.echo(click.style(str(exception), fg='red'), err=True)
-
-    # Mention also the original exception if provided
-    if exception.__cause__:
-        click.echo("The exception was caused by the previous exception:", err=True)
-        show_exception(exception.__cause__)
+# A small helper for "print to sys.stderr".
+OUT = functools.partial(print, file=sys.stderr)
```
Comment (thrix) [inline bin/tmt:12]: suggestion: if we are here, could we rename the helper function to `STDERR`? makes it more clear to me what it does as `OUT`

**Comment by thrix** [comment]
Also looks the tests are somehow unhappy, but those look unrelated to the change?

File: bin/tmt (hunk 1 above)
Comment (happz) [reply bin/tmt:12]: My original name was `print_error`, but then I saw `print_error()` calls (no arguments) which add some empty lines as a formatting help, and I was thiking, it's called "print_error" but it's called without any error, that's weird... So I went for `OUT` which is more related to `print`, and seeing it without any arguments wasn't that puzzling. So I'd be worried that `STDERR()` could be also suspicious, although not as much as the original "print error".

Comment (thrix) [reply bin/tmt:12]: yeah, naming :) it was just a suggestion, I leave it to other reviewers to share the opinion

Comment (happz) [reply bin/tmt:12]: Right, a missing final exception in the log. Good point, but would you mind if I add it in a follow-up patch?

Comment (lukaszachy) [reply bin/tmt:12]: Sure. This PR keeps status quo.

**Comment by lukaszachy** [comment]
> Also looks the tests are somehow unhappy, but those look unrelated to the change?

Randomly chosen `/tests/prepare/require` fails on 
```
:: [ 10:17:43 ] :: [   PASS   ] :: Command 'tmt run -avr provision -h container -i fedora plan --name missing | tee output' (Expected 2, got 2)
:: [ 10:17:43 ] :: [   FAIL   ] :: File 'output' should contain 'Unable to find a match: forest' 
```

Because previously it got printed to stdout and now is no stderr (and rlRun uses tee)

**Comment by happz** [comment]
> > Also looks the tests are somehow unhappy, but those look unrelated to the change?
> 
> Randomly chosen `/tests/prepare/require` fails on
> 
> ```
> :: [ 10:17:43 ] :: [   PASS   ] :: Command 'tmt run -avr provision -h container -i fedora plan --name missing | tee output' (Expected 2, got 2)
> :: [ 10:17:43 ] :: [   FAIL   ] :: File 'output' should contain 'Unable to find a match: forest' 
> ```
> 
> Because previously it got printed to stdout and now is to stderr (and rlRun uses tee)

Makes perfect sense, yet it's green on my laptop :/ I'll investigate.

**Comment by happz** [comment]
`how=full` complement results:

```
                                errr /tests/provision/virtual/dependencies (timeout)
                                pass /tests/core/adjust
                                pass /tests/core/debug
                                pass /tests/core/docs
                                pass /tests/core/dry
                                pass /tests/core/enabled
                                pass /tests/core/env
                                pass /tests/core/environment-file
                                pass /tests/core/error
                                pass /tests/core/fmf-id
                                pass /tests/core/force
                                pass /tests/core/link
                                pass /tests/core/ls
                                pass /tests/core/order
                                pass /tests/core/path
                                fail /tests/core/web-link
                                pass /tests/run/shell
```

Both red tests are expected, I have a custom ssh configuration, which affects `web-link` generation, and I re-ran the timeouting test manually.

File: bin/tmt (hunk 1 above)
Comment (psss) [reply bin/tmt:12]: Hm, naming is hard ;) I must admit that `OUT` is super confusing to me: At the first sight I think to myself what a weird constant name? And it's called? It's a function? Why function in capital letters? And what is it outing? ...

@happz, your first thought seems much better to me. Or, perhaps, what about `print_stderr()`?

**Comment by psss** [comment]
I've tried how the output should look like using:

```python
raise tmt.utils.PrepareError("I don't know!") from Exception("Something weird.")
```

The output looks like this:

```
    prepare
        how: install
        order: 50

I don't know!

The exception was caused by the previous exception:


Something weird.
```

Are the two empty lines expected?

**Comment by happz** [comment]
> I've tried how the output should look like using:
> 
> ```python
> raise tmt.utils.PrepareError("I don't know!") from Exception("Something weird.")
> ```
> 
> The output looks like this:
> 
> ```
>     prepare
>         how: install
>         order: 50
> 
> I don't know!
> 
> The exception was caused by the previous exception:
> 
> 
> Something weird.
> ```
> 
> Are the two empty lines expected?

Nope, removed in 5b37c44a.

File: bin/tmt (hunk 1 above)
Comment (happz) [reply bin/tmt]: Luckily for us, while naming is hard, renaming is pretty cheap. Changed back to `print_error()`, waiting eagerly for reviews.

**Review by thrix — APPROVED** [review]
LGTM, I am happy now

**Review by janhavlin — APPROVED** [review]

**Review by psss — APPROVED** [review]

**Pull request merged**